Docs: SCIM Sailpoint (#56759)
gravitational/teleport: docs/pages/admin-guides/management/guides/scim-sailpoint.mdx

---
title: SCIM Sailpoint Integration
description: How to manage Access List membership via SCIM SailPoint integration

---

> The SCIM integration between **SailPoint** and **Teleport** enables automated
> synchronizing SailPoint group memberships with
> Teleport Access Lists. This integration supports centralized identity governance in
> SailPoint while Teleport enforces fine-grained access controls.

Reviewer: ```suggestion
The SCIM integration between **SailPoint** and **Teleport** enables automated
synchronization of SailPoint group memberships and
Teleport Access Lists. This integration supports centralized identity governance in
SailPoint while Teleport enforces fine-grained access controls.
```

Grammar suggestion

User permissions in Teleport are defined through Access Lists. While role definitions

> live in Teleport, **group membership is dynamically managed** by SailPoint SCIM group

Reviewer: ```suggestion
live in Teleport, group membership is dynamically managed by SailPoint SCIM group
```
I don't think we need bold text here, since the bolded text makes up a substantial part of the sentence, and we tend to only bold introductions of new keywords.

assignments. This ensures users have up-to-date access aligned with organizational
policies.

## Prerequisites
- Teleport Enterprise v(=teleport.version=)

> - Teleport Terraform provider v(=teleport.version=)

Reviewer: I think we should link to the section of the docs that explains how to set up the provider: https://goteleport.com/docs/admin-guides/infrastructure-as-code/terraform-provider/

> - A running Teleport cluster with SSO connect (e.g., Okta SAML enabled

Reviewer: ```suggestion
- A running Teleport cluster with an SSO authentication connector (e.g., Okta SAML) enabled
```
- Grammar fix
- Clarifying a Teleport term, authentication connector, to be more consistent with the rest of the docs

- Access to SailPoint IdentityNow or IdentityIQ

> ### How it works

Reviewer: We've been standardizing the docs to make this section an H2 and place it before the Prerequisites to set the stage for the instructions with an architectural overview.

> The SCIM integration uses a **1:1 mapping** between SailPoint groups and Teleport

Reviewer: ```suggestion
The SCIM integration uses a 1:1 mapping between SailPoint groups and Teleport
```
Since we clarify this below, I don't think we need bold here—it's not the kind of text we need to use special formatting to make sure the reader notices.

Access Lists:

- Each SailPoint group’s `displayName` must match the `spec.title` of a Teleport
  Access List.
- Access Lists must be created in advance in Teleport and set to `type: "scim"`.
- Role assignments are handled in Teleport, while group membership is managed by
  SailPoint.

> ### Step 1/3. Configure SCIM Integration

Reviewer: Some "Step..." headings use the H2 level while some use H3. We should use H2 for all of them.

Teleport provides a guided web UI based configuration flow for SCIM integration.

1. In the Teleport Web UI, go to **"Add new integration"** and select **SCIM**.

   ![Plugin Install](../../../../img/sailpoint/plugin-install.png)

2. Select the **SAML connector** to associate SCIM-provisioned users with SSO logins.

   - By default, SSO users in Teleport are ephemeral.
   - SCIM provisioning ensures users are persistently created and managed by
     SailPoint.

   ![Connector Selection](../../../../img/sailpoint/plugin-connector.png)

3. Click **Continue** to proceed to the **SCIM Credentials** screen.

   - Teleport uses OAuth 2.0 Bearer Tokens for SCIM authentication.
   - Copy the **Client ID**, **Client Secret**, and **Base URL** — you'll use them
     when configuring SailPoint.

## Step 2/3: Create a SCIM-Managed Access List in Teleport

Create a new Access List in Teleport using Terraform. Be sure to set `type = "scim"`
and match `spec.title` to the SailPoint group name:

> ```hlc

Reviewer: ```suggestion
```hcl
```

resource "teleport_access_list" "acl-group-editor" {
  header = {
    version = "v1"
    name = "scim-group-editor"
    metadata = {
      labels = {
        scim = "sailpoint"
      }
    }
  }
  spec = {
    title = "GroupEditor"
    type = "scim"
    grants = {
      roles  = ["editor"]
      traits = []
    }
    owners = [
      {
        name = "alice"
      }
    ]
    membership_requires = {
      roles = []
    }
    ownership_requires = {
      roles = []
    }
    audit = {
      recurrence = {
        frequency    = 3
        day_of_month = 15
      }
    }
  }
}
```

<Admonition type="info">
    The SCIM group name (displayName) in SailPoint must exactly match

>     spec.title in the Teleport Access List.

Reviewer: ```suggestion
    `spec.title` in the Teleport Access List.
```

</Admonition>

### Step 3/4: Configure SCIM 2.0 Teleport Connector in SailPoint

To integrate Teleport with SailPoint using SCIM, you need to configure a SCIM connector in SailPoint IdentityNow or SailPoint IdentityIQ.
The exact configuration steps may vary slightly depending on your version of SailPoint, but the general process is as follows:

> 1. Configure SCIM in SailPoint

Reviewer: As an alternative to the two-level outline, which I think can be difficult to read, we can make each item in the outer level an H3 heading, and turn the inner level into a one-level list, e.g.,:

```markdown
### Configure SCIM in SailPoint

1. Go to **Applications > Application Definition > Add New Application**.

2. Select **SCIM 2.0** as the application type and provide the required configuration details
```
